src/lib/util/flag-resolver.ts

```diff
@@ -22,13 +23,16 @@ export default class FlagResolver implements IFlagResolver {
         const flags: IFlags = { ...this.experiments };
 
         Object.keys(flags).forEach((flagName: IFlagKey) => {
-            if (!this.experiments[flagName]) {
-                if (typeof flags[flagName] === 'boolean') {
+            const flag = flags[flagName];
+            if (typeof flag === 'boolean') {
+                if (!flag) {
                     flags[flagName] = this.externalResolver.isEnabled(
                         flagName,
                         context,
                     );
-                } else {
+                }
+            } else {
+                if (!flag?.enabled) {
```

Comment:
Should we still resolve `getVariant` even if it's not enabled? It would give us the default variant

Reply:
Same as [above](https://github.com/Unleash/unleash/pull/3808#discussion_r1197983537), if the initial value is `falsy` (not overridden be either `experimental` or `server-dev` to be `truthy`), we want to see what our external resolver has to say about it.

It may not be the default variant in this case. We may have a `falsy` initial value, but our external resolver (e.g. Unleash hosted) saying that this flag should be enabled, etc.